src/utils/hooks/useGetSectionProgress.tsx

```diff
@@ -0,0 +1,86 @@
+import { useState, useEffect } from 'react'
+import { getFirstErrorLocation, revalidateAll } from '../helpers/application/revalidateAll'
+import {
+  ApplicationElementStates,
+  ResponsesByCode,
+  SectionsProgress,
+  TemplateSectionPayload,
+  User,
+} from '../types'
+
+interface UseGetSectionProgressProps {
+  loadStart: boolean
+  currentUser: User
+  serialNumber: string
+  templateSections: TemplateSectionPayload[]
+  elementsState: ApplicationElementStates
+  responsesByCode: ResponsesByCode
+}
+
+const useGetSectionsProgress = ({
+  loadStart,
+  currentUser,
+  serialNumber,
+  templateSections,
+  elementsState,
+  responsesByCode,
+}: UseGetSectionProgressProps) => {
+  const [sections, setSections] = useState<SectionsProgress>()
+  const [isLoadingProgress, setIsLoadingProgress] = useState(false)
+
+  useEffect(() => {
+    if (loadStart && currentUser && templateSections && elementsState && responsesByCode) {
+      setIsLoadingProgress(true)
+      runScanSectionProgress()
+    }
+  }, [loadStart])
+
+  async function runScanSectionProgress() {
+    let sectionsProgress: SectionsProgress = {}
+    templateSections.forEach(async ({ code, title, index }) => {
```

Comment:
As discussed, you probably want a `for...in` loop here as it's in an `async` function